### PR TITLE
Opcode Metadata

### PR DESCRIPTION
**Pull request by ike709**
Adds a custom attribute for specifying metadata on individual opcodes.

Used this to pretty much nuke `ShrinkStack()` and `GrowStack()`.

Please pay close attention to the attributes because this is a great time to audit MaxStack.

### CODE REVIEW

**Comment by ike709** [comment]
It may be trivial to handle local non-static vars properly depending on how wixoa answers a question on Discord.

**Comment by ike709** [comment]
Locals should now be handled.

**Comment by ike709** [comment]
Nuked proc purity from this PR to do it differently in a future PR. I still think the metadata system is useful to keep.

File: DMCompiler/Bytecode/DreamProcOpcode.cs [hunk 1]
```diff
@@ -402,3 +402,50 @@ public static class OpcodeVerifier {
...
+{
+    private static readonly OpcodeMetadata[] MetadataCache = new OpcodeMetadata[256];
+
+    static OpcodeMetadataCache()
+    {
+        foreach (DreamProcOpcode opcode in Enum.GetValues(typeof(DreamProcOpcode)))
+        {
+            var field = typeof(DreamProcOpcode).GetField(opcode.ToString());
+            var attribute = Attribute.GetCustomAttribute(field!, typeof(OpcodeMetadataAttribute));
+            var metadataAttribute = Unsafe.As<OpcodeMetadataAttribute>(attribute);
```
Comment (wixoaGit) [inline DMCompiler/Bytecode/DreamProcOpcode.cs:442]: Why the `Unsafe.As<>()`? a normal cast should work fine

File: DMCompiler/DM/DMProc.cs [hunk 2]
```diff
@@ -111,6 +111,7 @@ private class DMProcScope {
         private readonly List<SourceInfoJson> _sourceInfo = new();
         private string? _lastSourceFile;
 
+
```
Comment (wixoaGit) [inline DMCompiler/DM/DMProc.cs:114]: Extra line
```suggestion
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: DMCompiler/Bytecode/DreamProcOpcode.cs (hunk 1 above)
Comment (ike709) [reply DMCompiler/Bytecode/DreamProcOpcode.cs:439]: Why not?

Comment (wixoaGit) [reply DMCompiler/Bytecode/DreamProcOpcode.cs:439]: It's generally good to stay away from unsafe things unless necessary.

While `attribute` will probably always be an `OpcodeMetadataAttribute` here, you're still foregoing any safety checks which could lead to really odd issues down the line.

Comment (ike709) [reply DMCompiler/Bytecode/DreamProcOpcode.cs:439]:
```suggestion
            var metadataAttribute = (OpcodeMetadataAttribute)attribute;
```

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**